src/api/helpers/entityLoader.js

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2019  Akhilesh Kumar
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+// @flow
+
+
+import * as commonUtils from '../../common/helpers/utils';
+
+/**
+ * This is a middleware function to load entity detail according to given relations
+ *
+ * @param {string} modelName - Name of entity model
+ * @param {string[]} relations - List of entity relations for fetching the related detail
+ * @param {string} errMessage - Error message, if any error will occur
+ * @returns {object} an object containing the error message if any error will occur.
+ * If entity is founded succesfully in the database this function set the entity data
+ * at res.locals.entity and return to next function.
+ * @example
+ *		const errorMessage = 'Edition  is not founded';
+ *		makeEntityLoader('Edition', ['defaultAlias.language', 'editionStatus'], errMessage);
+ * First, check the bbid is valid or not.
```

Comment:
This would be better in an `@description` block rather than in the `@example` block.

Reply:
And by "this", I mean lines 36 to 41